src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -365,12 +372,15 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
             : vizConfig.valueSpecConfig,
         // set undefined for variable change
         checkedLegendItems: undefined,
-        // set independentAxisRange undefined
-        independentAxisRange: undefined,
+        independentAxisRange: keepIndependentAxisRange
+          ? vizConfig.independentAxisRange
+          : undefined,
         dependentAxisRange: undefined,
         independentAxisLogScale: false,
         dependentAxisLogScale: false,
-        independentAxisValueSpec: 'Full',
+        independentAxisValueSpec: keepIndependentAxisRange
```

Comment:
Are we not going to keep `independentAxisLogScale` here too?  (two lines above)
Might be best to use the same name for the variable: `keepIndependentAxisSettings`.

Reply:
@bobular good point! 👍 I think it can also keep its setting so will change it. Also, I realized that I did make a wrong name for the replacement of keepBin - will fix it too.

Reply:
@bobular changes are made!